macros/recursive_dag.sql

```diff
@@ -4,8 +4,28 @@
 
 {% macro default__recursive_dag() %}
 
-with recursive direct_relationships as (
-    select  
+with recursive direct_relationships
+    {%- if target.type == 'athena' %}
```

Comment:
To keep the default `recursive_dag` clean, can you create a new `athena__recursive_dag` macro for the athena-specific logic (similar to [bigquery__recursive_dag](https://github.com/dbt-labs/dbt-project-evaluator/blob/601263c23127efb53678110d45c5c5148b44ec1d/macros/recursive_dag.sql#L137))?

Reply:
Thanks for the suggestion, I thought about that as well. However adding extra implementation leads to need to support 2 macroses and someone may forget to update all implementations in case of change in default apapter. So as for Athena, we don't need to change sql query but just add required fields list. WDYT @graciegoheen Is it critical to add extra implementaion or we can leave it as is?

Reply:
I agree with Grace, the dispatched macros are entirely designed to handle the different logic for different adapter types.

One of the main advantage is that they support "inheritance", so if in the future there is another adapter that is a "child" of the athena one with just some slight differences, it will inherit all the `athena__xxx` macros despite having a different `target.type`.

Reply:
fixed